airbyte-integrations/connectors/destination-astra/destination_astra/destination.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any, Iterable, Mapping
+
+from airbyte_cdk import AirbyteLogger
+from airbyte_cdk.destinations import Destination
+from airbyte_cdk.destinations.vector_db_based.document_processor import DocumentProcessor
+from airbyte_cdk.destinations.vector_db_based.embedder import Embedder, create_from_config
+from airbyte_cdk.destinations.vector_db_based.indexer import Indexer
+from airbyte_cdk.destinations.vector_db_based.writer import Writer
+from airbyte_cdk.models import AirbyteConnectionStatus, AirbyteMessage, ConfiguredAirbyteCatalog, ConnectorSpecification, Status
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+from destination_astra.config import ConfigModel
+from destination_astra.indexer import AstraIndexer
+
+BATCH_SIZE = 100
+
+
+class DestinationAstra(Destination):
+    indexer: Indexer
+    embedder: Embedder
+
+    def _init_indexer(self, config: ConfigModel):
+        self.embedder = create_from_config(config.embedding, config.processing)
+        self.indexer = AstraIndexer(config.indexing, self.embedder.embedding_dimensions)
+
+    def write(
+        self, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog, input_messages: Iterable[AirbyteMessage]
+    ) -> Iterable[AirbyteMessage]:
+        config_model = ConfigModel.parse_obj(config)
+        self._init_indexer(config_model)
+        writer = Writer(
+            config_model.processing, self.indexer, self.embedder, batch_size=BATCH_SIZE, omit_raw_text=config_model.omit_raw_text
+        )
+        yield from writer.write(configured_catalog, input_messages)
+
+    def check(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> AirbyteConnectionStatus:
+        parsed_config = ConfigModel.parse_obj(config)
+        self._init_indexer(parsed_config)
```

Comment:
On https://github.com/airbytehq/airbyte/pull/34058/files#r1459806142 you wrote you moved the `create_collection` call, but it seems like some merge foo brought it back (it's also not in the indexer). Could you double check please?

Reply:
Its on the next line here

Reply:
@anomnaco yes, it shouldn’t be here in the destination class, it should be in the indexer class check method so it’s properly wrapped in a try/except block and a meaningful error message is produced on check

Reply:
That caused issues with the unit and integration tests

Reply:
Hi @flash1293 - I've been chatting with Obi about this offline. He was testing with this in the indexer class check method and got failures with the unit tests with it there. Do you need us to look deeper into getting this to work in the way that you suggested or is it acceptable here?

Reply:
We are working on resolving this now.

Reply:
The important point to me is that the user gets an actionable error message on check as part of a well-formed status message and the connector doesn't crash as this won't produce a nice output in the UI.